osu.Game/Screens/Ranking/Expanded/Accuracy/GradedCircles.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Game.Graphics;
+using osu.Game.Scoring;
+
+namespace osu.Game.Screens.Ranking.Expanded.Accuracy
+{
+    public partial class GradedCircles : CompositeDrawable
+    {
+        private double progress;
+
+        public double Progress
+        {
+            get => progress;
+            set
+            {
+                progress = value;
+                dProgress.RevealProgress = value;
+                cProgress.RevealProgress = value;
+                bProgress.RevealProgress = value;
+                aProgress.RevealProgress = value;
+                sProgress.RevealProgress = value;
+                xProgress.RevealProgress = value;
```

Comment:
This is pretty terrible. Can you put the `GradedCircle` instances in a `Container<GradedCircle>` and iterate over that instead?